src/main/java/org/folio/rest/impl/UsersAPI.java

```diff
@@ -640,6 +642,49 @@ public void getUsersProfilePictureByProfileId(String profileId, Map<String, Stri
       });
   }
 
+  @Override
+  @Stream
+  public void putUsersProfilePictureByProfileId(String profileId, InputStream entity, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    try (InputStream bis = new BufferedInputStream(entity)) {
+      if (Objects.isNull(okapiHeaders.get(STREAM_ABORT))) {
+        validateAndProcessByteArray(bis);
+        if (Objects.nonNull(requestBytesArray) && requestBytesArray.length != 0) {
+          if (ProfilePictureHelper.detectFileType(requestBytesArray).equals("Unknown")) {
+            asyncResultHandler.handle(
+              succeededFuture(PutUsersProfilePictureByProfileIdResponse.respond500WithApplicationJson("Requested image should be of supported type-[PNG,JPG,JPEG]")));
+          }
+          Buffer binaryDataBuffer = Buffer.buffer(requestBytesArray);
+          PgUtil.postgresClient(vertxContext, okapiHeaders)
+            .execute(createUpdateQuery(okapiHeaders), Tuple.of(binaryDataBuffer, profileId))
+            .compose(rows -> {
+              if(rows.rowCount() != 0) {
+                return succeededFuture(PutUsersProfilePictureByProfileIdResponse.respond200WithApplicationJson(mapResultSetToProfilePicture(rows)));
+              } else {
+                return succeededFuture(PutUsersProfilePictureByProfileIdResponse.respond404WithTextPlain("Existing profile picture is not found"));
+              }
+            })
+            .map(Response.class::cast)
+            .onComplete(reply -> {
+              if (reply.cause() != null) {
+                asyncResultHandler.handle(
+                  succeededFuture(PutUsersProfilePictureByProfileIdResponse.respond500WithApplicationJson(reply.cause().getMessage())));
+              }
+              asyncResultHandler.handle(reply);
+            });
+        } else {
```

Comment:
The complexity of the method is growing because different conditional checks can we think of breaking this method to have some smaller methods 
1. check file type 
2. Size of the file 
3. profile picture already exists
4. 500 errors.
5. actual persisting change for the PUT API

Reply:
@gurleenkaurbp These review comments will be taken care while we implement the object storage implementation of Put api